src/CommunityToolkit.Maui.MediaElement/Views/MediaManager.android.cs

```diff
@@ -411,6 +412,23 @@ public void OnVolumeChanged(float volume)
 		PlayerView.KeepScreenOn = MediaElement.ShouldKeepScreenOn;
 	}
 
+	protected virtual partial void PlatformUpdateShouldMute()
+	{
+		if (Player is null || MediaElement is null)
+		{
+			return;
+		}
+
+		// We're going to muted state, capture current volume first
+		// so we can restore later
+		if (MediaElement.ShouldMute == true)
+		{
+			volumeBeforeMute = Player.Volume;
+		}
+
+		Player.Volume = MediaElement.ShouldMute ? 0 : volumeBeforeMute;
```

Comment:
can the Volume be changed while on mute?
like this:
```
MediaElement.Volume = 50; // volume 50, volumeBeforeMove 1
MediaElement.ShouldMute = true; // volume 0, volumeBeforeMove 50
MediaElement.Volume = 100; // volume 100, volumeBeforeMove 50
MediaElement.ShouldMute = false; // volume 50, volumeBeforeMove 50

Expected volume 100

Reply:
Hmm yeah I guess. I hate it when I am implementing features that should be available in these controls... :P I will think about this some more. Thanks!

Reply:
Oh so... I think the solution is actually the other way around. If you start updating the volume it will probably unmute. So the solution here is when the volume is updated, the `ShouldMute` could become false again I guess.

Reply:
You can update volumeBeforeMute when Volume changes, except this line of code

Reply:
Whichever path we move forward with, let's ensure our implementation is consistent across iOS, Android, macOS, Windows + Tizen

Ie `MediaElement` shouldn't unmute when the volume changes on Android and not unmute when the volume changes on iOS